axlearn/cloud/gcp/dataflow_inference_custom.py

```diff
@@ -0,0 +1,205 @@
+"""An Apache Beam example pipeline to run batch inference jobs using a model trained with AXLearn.
+Command line options:
+--module: the same module used for training
+--config: the same config used for training
+--trainer_dir: location of your checkpoints for inference
+
+To debug locally:
+$ docker run -it --mount type=bind,src=$HOME/.config/gcloud,dst=/root/.config/gcloud \
+    --entrypoint /bin/bash ${DOCKER_REPO}/${DOCKER_IMAGE}:{DOCKER_TAG}
+> python3 -m axlearn.cloud.gcp.dataflow_inference_custom \
+    --module=text.gpt.c4_trainer \
+    --config=fuji-7B-single  \
+    --trainer_dir='gs://.../checkpoints/step_xxx'
```

Comment:
nit -- Usually, `trainer_dir` refers to the root of this dir. Maybe we can call it `checkpoint_dir`?

Reply:
I think these flags need to be the same as the trainer config flags. However, during my testing, even though `trainer_dir` here goes into the step folder, this function that eventually takes the flags still works fine: https://github.com/apple/axlearn/blob/main/axlearn/common/launch_trainer.py#L69